### PR TITLE
Remove unstable queries from performance/join_set_filter

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

Test failed here:
https://s3.amazonaws.com/clickhouse-test-reports/0/3f892ceb12b868b1b9deb0607a7df2dde5f1a139/performance_comparison_[1_4]/report.html

Let's check execution time for each query in this test:

```
ubuntu@ip-172-31-42-13~/_logs $ grep 'join_set_filter.query.*.prewarm' right-server-log.log left-server-log.log | grep 'Processed in'
right-server-log.log:2023.05.24 14:30:04.612129 [ 512 ] {join_set_filter.query0.prewarm0} <Debug> TCPHandler: Processed in 0.226517489 sec.
right-server-log.log:2023.05.24 14:31:16.962484 [ 512 ] {join_set_filter.query5.prewarm0} <Debug> TCPHandler: Processed in 33.313428802 sec.
right-server-log.log:2023.05.24 14:31:45.272076 [ 512 ] {join_set_filter.query7.prewarm0} <Debug> TCPHandler: Processed in 0.493587454 sec.
right-server-log.log:2023.05.24 14:31:52.242882 [ 512 ] {join_set_filter.query1.prewarm0} <Debug> TCPHandler: Processed in 0.335373937 sec.
right-server-log.log:2023.05.24 14:31:55.943563 [ 512 ] {join_set_filter.query3.prewarm0} <Debug> TCPHandler: Processed in 0.469933564 sec.
right-server-log.log:2023.05.24 14:32:00.771681 [ 512 ] {join_set_filter.query2.prewarm0} <Debug> TCPHandler: Processed in 0.4107554 sec.
right-server-log.log:2023.05.24 14:32:05.677168 [ 512 ] {join_set_filter.query6.prewarm0} <Debug> TCPHandler: Processed in 1.001121192 sec.
right-server-log.log:2023.05.24 14:35:42.602096 [ 512 ] {join_set_filter.query4.prewarm0} <Debug> TCPHandler: Processed in 180.01542621 sec.
left-server-log.log:2023.05.24 14:30:04.385101 [ 509 ] {join_set_filter.query0.prewarm0} <Debug> TCPHandler: Processed in 0.637355368 sec.
left-server-log.log:2023.05.24 14:30:43.648150 [ 509 ] {join_set_filter.query5.prewarm0} <Debug> TCPHandler: Processed in 36.00816096 sec.
left-server-log.log:2023.05.24 14:31:44.707640 [ 509 ] {join_set_filter.query7.prewarm0} <Debug> TCPHandler: Processed in 0.895852271 sec.
left-server-log.log:2023.05.24 14:31:51.902449 [ 509 ] {join_set_filter.query1.prewarm0} <Debug> TCPHandler: Processed in 0.254638137 sec.
left-server-log.log:2023.05.24 14:31:55.472064 [ 509 ] {join_set_filter.query3.prewarm0} <Debug> TCPHandler: Processed in 0.244279989 sec.
left-server-log.log:2023.05.24 14:32:00.360105 [ 509 ] {join_set_filter.query2.prewarm0} <Debug> TCPHandler: Processed in 1.472849179 sec.
left-server-log.log:2023.05.24 14:32:04.668561 [ 509 ] {join_set_filter.query6.prewarm0} <Debug> TCPHandler: Processed in 0.814490698 sec.
left-server-log.log:2023.05.24 14:32:42.581501 [ 509 ] {join_set_filter.query4.prewarm0} <Debug> TCPHandler: Processed in 30.493933458 sec.
```

Prewarm for query4 and query5 are the longest. On one server it even too >180 sec end was killed by timeout:

[full logs for queries](https://pastila.nl/?06154704/cf5c0ed0b5a96ef5b8441a0e2c3007bc)

```
left-server-log.log:2023.05.24 14:32:12.833552 [ 1265 ] {join_set_filter.query4.prewarm0} <Debug> FilterBySetOnTheFlyTransform: Finished create set and filter right joined stream: bypass 7240966 rows
left-server-log.log:2023.05.24 14:32:40.478315 [ 1265 ] {join_set_filter.query4.prewarm0} <Debug> MergingSortedTransform: Merge sorted 16 blocks, 999908 rows in 28.384107487 sec., 35227.74145560013 rows/sec., 559.19 KiB/sec
left-server-log.log:2023.05.24 14:32:42.577095 [ 1260 ] {join_set_filter.query4.prewarm0} <Debug> MergingSortedTransform: Merge sorted 1529 blocks, 100000000 rows in 30.484114273 sec., 3280397.098123029 rows/sec., 50.16 MiB/sec
left-server-log.log:2023.05.24 14:32:42.577947 [ 1260 ] {join_set_filter.query4.prewarm0} <Trace> MergeJoinAlgorithm: Finished pocessing in 30.484114273 seconds, left: 16 blocks, 999908 rows; right: 1529 blocks, 100000000 rows, max blocks loaded to memory: 3
left-server-log.log:2023.05.24 14:32:42.580244 [ 509 ] {join_set_filter.query4.prewarm0} <Debug> executeQuery: Read 200000000 rows, 2.98 GiB in 30.491888 sec., 6559121.560462245 rows/sec., 100.08 MiB/sec.
left-server-log.log:2023.05.24 14:32:42.581488 [ 509 ] {join_set_filter.query4.prewarm0} <Debug> MemoryTracker: Peak memory usage (for query): 1.56 GiB.
left-server-log.log:2023.05.24 14:32:42.581501 [ 509 ] {join_set_filter.query4.prewarm0} <Debug> TCPHandler: Processed in 30.493933458 sec.
```

```
right-server-log.log:2023.05.24 14:32:43.414046 [ 2975 ] {join_set_filter.query4.prewarm0} <Debug> FilterBySetOnTheFlyTransform: Finished create set and filter right joined stream: bypass 5111808 rows
right-server-log.log:2023.05.24 14:32:43.418308 [ 1328 ] {join_set_filter.query4.prewarm0} <Debug> FilterBySetOnTheFlyTransform: Finished create set and filter right joined stream: bypass 8224768 rows
right-server-log.log:2023.05.24 14:35:42.595315 [ 512 ] {join_set_filter.query4.prewarm0} <Error> executeQuery: Code: 159. DB::Exception: Timeout exceeded: elapsed 180.005253741 seconds, maximum: 180. (TIMEOUT_EXCEEDED) (version 23.5.1.1953 (official build)) (from [::1]:57524) (in query: SELECT * FROM t1 JOIN t2 ON t1.x = t2.x WHERE t1.y % 100 = 0), Stack trace (when copying this message, always include the lines below):
right-server-log.log:2023.05.24 14:35:42.602059 [ 512 ] {join_set_filter.query4.prewarm0} <Error> TCPHandler: Code: 159. DB::Exception: Timeout exceeded: elapsed 180.005253741 seconds, maximum: 180. (TIMEOUT_EXCEEDED), Stack trace (when copying this message, always include the lines below):
right-server-log.log:2023.05.24 14:35:42.602096 [ 512 ] {join_set_filter.query4.prewarm0} <Debug> TCPHandler: Processed in 180.01542621 sec.
right-server-log.log:2023.05.24 14:35:42.602130 [ 512 ] {join_set_filter.query4.prewarm0} <Debug> MemoryTracker: Peak memory usage (for query): 1.57 GiB.
```

Optimization `FilterBySetOnTheFlyTransform` that this tests should check not used in this query and most of the execution time takes reading the table and sorting them. I compared traces and profile event for query that executed successful (on left server) and one that was killed by timeout (on right) and haven't found anything suspicious.

[traces for query from left](https://pastila.nl/?00bfc54a/30fc1150fa65a701d9a0a8c37dd7d0a2)
[traces for query from right](https://pastila.nl/?00bfc54a/4e22de3125d0bf2087397a2e57e52541)
[profile events](https://pastila.nl/?01fb6ef0/dccafccb581ee66a383995517a4900ac)

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit bd505fa4b2ff09b2f6a8479aeb598fa27328b19c with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50235/bd505fa4b2ff09b2f6a8479aeb598fa27328b19c/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by SmitaRKulkarni — APPROVED** [review]
LGTM

**Comment by SmitaRKulkarni** [comment]
@vdimir : The performance test fails with 'Errors while building the report' are because this these queries exist in master & not in this branch right ?. Then I guess they are expected to fail.

**Comment by vdimir** [comment]
Failed performance comparison runs do not contain join_set_filter,  https://s3.amazonaws.com/clickhouse-test-reports/50235/bd505fa4b2ff09b2f6a8479aeb598fa27328b19c/performance_comparison_[2_4]/report.html

But one that contain it https://s3.amazonaws.com/clickhouse-test-reports/50235/bd505fa4b2ff09b2f6a8479aeb598fa27328b19c/performance_comparison_[1_4]/report.html succeeded

File: tests/performance/join_set_filter.xml [hunk 1]
```diff
@@ -34,9 +34,6 @@
     <query>SELECT * FROM t1 JOIN t2 ON t1.x = t2.x WHERE greater(t1.y, {table_size} - 10000)</query>
     <query>SELECT * FROM t2 JOIN t1 ON t1.x = t2.x WHERE greater(t1.y, {table_size} - 10000)</query>
```
Comment (vdimir) [inline tests/performance/join_set_filter.xml:36]:
```suggestion
    <query>SELECT * FROM t1 JOIN t2 ON t1.x = t2.x WHERE t1.y % 100 = 0 AND t1.y < 10000</query>
    <query>SELECT * FROM t2 JOIN t1 ON t1.x = t2.x WHERE t1.y % 100 = 0 AND t1.y < 10000</query>
```

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**